File: kotlin/src/main/kotlin/y2015/Day02.kt
package y2015

import utils.Day
import utils.collections.product
import utils.readers.asLines

class Day02 : Day<Int> {

    private val input = file.asLines().map { it.split("x").map { it.toInt() } }

    override fun runAll() = super.run({ partOne(input) }) { partTwo(input) }

    private fun partOne(input: List<List<Int>>): Int {
        var area = 0
        for (i in input) {
            area += listOf(
                i[0] * i[1],
                i[0] * i[2],
                i[1] * i[2]
            ).let {
                it.sum() * 2 + (it.minOrNull() ?: 0)
            }
        }
        return area
    }

    private fun partTwo(input: List<List<Int>>): Int {
        var area = 0
        for (i in input) {
            area += i.sorted().let {
                it[0] * 2 + it[1] * 2 + it.product()
            }
        }
        return area
    }
}

File: rust/src/lib/years/y2021/day02.rs
use crate::lib::utils::tday::TDay;

pub struct Day02 {
    pub file: String,
    pub desc: String,
}

impl TDay for Day02 {
    fn desc(&self) -> String {
        (&self.desc).to_string()
    }

    fn part_one(&self) -> String {
        format!("{}", 2)
    }

    fn part_two(&self) -> String {
        format!("{}", 2)
    }
}